### ISSUE TITLE
Cannot slide image inside Flatlist

### ISSUE THREAD

**Issue by sumudu-dewasurendra** [intergalacticspacehighway/react-native-reanimated-zoom#2]
When using react-native-reanimated-zoom component inside Flatlist it cannot slide. 

Platform - Android
Code example - 
```
<FlatList
          data={media}
          extraData={media}
          horizontal={true}
          renderItem={({ item, index }) => {
            <Zoom>
              <Image {...imageProps} source={{ uri: item.imageUri }}/>
            </Zoom>
          }}
          keyExtractor={(item, index) => `${index}`}
          showsVerticalScrollIndicator={false}
          showsHorizontalScrollIndicator={false}
          initialScrollIndex={focusIndex}
          onScrollToIndexFailed={() => {
          }}
          pagingEnabled={true}
          onViewableItemsChanged={onViewRef.current}
          viewabilityConfig={viewConfigRef.current}
          scrollEnabled={!isUsingMedia}
          getItemLayout={flatlistLayout}
        />
```

**Comment by intergalacticspacehighway** [comment]
I think the pan responder is taking over the FlatList swipes. I'll try something. In the meanwhile, can you try the below gist and let me know if it works as expected?

https://gist.github.com/intergalacticspacehighway/864d8d510c324a720ead72efac5f0551

**Comment by putuoka** [comment]
> I think the pan responder is taking over the FlatList swipes. I'll try something. In the meanwhile, can you try the below gist and let me know if it works as expected?
> 
> https://gist.github.com/intergalacticspacehighway/864d8d510c324a720ead72efac5f0551

cab u please help me to make that work in my react native javascript types? i'm not using typescript btw

**Comment by intergalacticspacehighway** [comment]
@putuoka can you try this?
https://gist.github.com/intergalacticspacehighway/bf00f613ffc61b1e4da504d55ef8e4d2

**Comment by putuoka** [comment]
> @putuoka can you try this? https://gist.github.com/intergalacticspacehighway/bf00f613ffc61b1e4da504d55ef8e4d2

thanks that was fast. it's working. no error but the behaviour is change. better the original one. the pinch always bounce back to normal because u add reset code 

```
        translationX.value = withTiming(0);
        translationY.value = withTiming(0);
        scale.value = withTiming(1);
        originX.value = 0;
        originY.value = 0;
        isPinching.value = false;
        prevScale.value = 0;
        prevTranslationX.value = 0;
        prevTranslationY.value = 0;
        panTranslateX.value = 0;
        panTranslateY.value = 0;
```

also i tried new prop onPinchStart but my console log always run even i'm not using or start pinch

```
<GestureHandlerRootView>
      <PinchToZoom onPinchStart={console.log('test')}>
       <RenderComponent item={item} />
        </PinchToZoom>
</GestureHandlerRootView>
```

**Comment by intergalacticspacehighway** [comment]
Can you try 0.1.3 and let me know how it works?

**Comment by intergalacticspacehighway** [comment]
added an [API](https://github.com/intergalacticspacehighway/react-native-reanimated-zoom#with-flatlist-or-scrollview) in 0.2.0 that handles a few more cases!

**Comment by intergalacticspacehighway** [comment]
Closing as it's solved in 0.2.0. Let me know if you still face any issues.
